pkg/controller/infrastructure/infraflow/ensurer.go

```diff
@@ -699,6 +741,88 @@ func (fctx *FlowContext) EnsureManagedIdentity(ctx context.Context) (err error)
 	return err
 }
 
+// MigrateAvailabilitySet prepares an AS-based shoot to be migrated to VMSS-Flex.
+func (fctx *FlowContext) MigrateAvailabilitySet(ctx context.Context) error {
+	var (
+		log = shared.LogFromContext(ctx)
+		c   = fctx.client
+	)
+
+	// return early if the migration has already been complete
+	if v := fctx.whiteboard.GetChild("migration").GetChild(KindAvailabilitySet.String()).Get("complete"); v != nil && *v == "true" {
+		return nil
+	}
+	// return early if the cluster does not have AS.
+	if fctx.whiteboard.GetChild(ChildKeyIDs).Get(KindAvailabilitySet.String()) == nil {
+		return nil
+	}
+
+	// IF VMOs are not needed, or the migration is already done, return early.
+	if !helper.HasShootVmoMigrationAnnotation(fctx.cluster.Shoot.GetAnnotations()) {
+		return nil
+	}
+
+	log.Info("Preparing for the migration to VMOs")
+	scaleDownDeployment := func(ctx context.Context, key k8sclient.ObjectKey) error {
+		log.Info("Scaling deployment to 0 replicas", "Name", key.String())
+		deployment := &appsv1.Deployment{}
+		if err := fctx.client.Get(ctx, k8sclient.ObjectKey{
+			Namespace: fctx.infra.Namespace,
+			Name:      azure.CloudControllerManagerName,
+		}, deployment); k8sclient.IgnoreNotFound(err) != nil {
+			return err
+		} else if k8sclient.IgnoreNotFound(err) != nil {
+			return nil
+		}
```

Comment:
```suggestion
		if err := fctx.client.Get(ctx, k8sclient.ObjectKey{
			Namespace: fctx.infra.Namespace,
			Name:      azure.CloudControllerManagerName,
		}, deployment); k8sclient.IgnoreNotFound(err) != nil {
			return err
		}
```
Not sure if this is the right one to keep